tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingJobExecution.scala

```diff
@@ -17,21 +18,21 @@ import java.util.regex.Pattern
  * @param mappingTasks               List of mapping tasks to be executed (as a subset of the mapping tasks defined in the job)
  * @param jobGroupIdOrStreamingQuery Keeps Spark job group id for batch jobs and StreamingQuery for streaming jobs
  */
-// TODO: The FhirMappingJobExecution model currently includes the entire FhirMappingJob ('job' field), which is unnecessary.
-//  We should remove the 'job' field from the model. Instead, add only the necessary fields to the model.
-case class FhirMappingJobExecution(id: String = UUID.randomUUID().toString,
-                                   projectId: String = "",
-                                   job: FhirMappingJob,
-                                   mappingTasks: Seq[FhirMappingTask] = Seq.empty,
-                                   jobGroupIdOrStreamingQuery: Option[Either[String, collection.mutable.Map[String, StreamingQuery]]] = None
+case class FhirMappingJobExecution(id: String,
```

Comment:
The new constructor is good but it can be improved.

There are two different usages of sourceSettings: 

1. sourceSettings.head._2.isInstanceOf[FileSystemSourceSettings]
2. sourceSettings.exists(source => source._2.asStream)

Therefore, instead of passing the all source settings, you can define two parameters: isStreamingJob and hasFileSystemSource. Then, please remove isStreaming method which becomes redundant.

Futher, please add the javadoc for all parameters.